src/HearThis/Communication/AndroidSynchronization.cs

```diff
@@ -47,20 +48,40 @@ public static void DoAndroidSync(Project project)
 			dlg.ShowAndroidIpAddress(); // AFTER we set our IP address, which may be used to provide a default
 			dlg.GotSync += (o, args) =>
 			{
-				var theirLink = new AndroidLink();
-				// Enhance: some way to validate that we really got an IP address.
-				theirLink.AndroidAddress = AndroidSyncDialog.AndroidIpAddress;
-				var ourLink = new WindowsLink(Program.ApplicationDataBaseFolder);
-				var merger = new RepoMerger(project, ourLink, theirLink);
-				merger.Merge(dlg.ProgressBox);
-				//Update info.txt on Android
-				var infoFilePath = project.GetProjectRecordingStatusInfoFilePath();
-				File.WriteAllText(infoFilePath, project.GetProjectRecordingStatusInfoFileContent());
-				var theirInfoTxtPath = project.Name + "/" + Project.InfoTxtFileName;
-				theirLink.PutFile(theirInfoTxtPath, File.ReadAllBytes(infoFilePath));
-				theirLink.SendNotification("syncCompleted");
-				dlg.ProgressBox.WriteMessage("Sync completed successfully");
-				//dlg.Close();
+				try
+				{
+					var theirLink = new AndroidLink();
+					// Enhance: some way to validate that we really got an IP address.
+					theirLink.AndroidAddress = AndroidSyncDialog.AndroidIpAddress;
+					var ourLink = new WindowsLink(Program.ApplicationDataBaseFolder);
+					var merger = new RepoMerger(project, ourLink, theirLink);
+					merger.Merge(dlg.ProgressBox);
+					//Update info.txt on Android
+					var infoFilePath = project.GetProjectRecordingStatusInfoFilePath();
+					File.WriteAllText(infoFilePath, project.GetProjectRecordingStatusInfoFileContent());
+					var theirInfoTxtPath = project.Name + "/" + Project.InfoTxtFileName;
+					theirLink.PutFile(theirInfoTxtPath, File.ReadAllBytes(infoFilePath));
+					theirLink.SendNotification("syncCompleted");
+					dlg.ProgressBox.WriteMessage("Sync completed successfully");
+					//dlg.Close();
+				}
+				catch (WebException ex)
+				{
+					if (ex.Status == WebExceptionStatus.NameResolutionFailure)
+					{
+						dlg.ProgressBox.WriteError("HearThis could not make sense of the address you gave for the device. Please try again.");
+					} else if (ex.Status == WebExceptionStatus.ConnectFailure)
+					{
+						dlg.ProgressBox.WriteError("HearThis could not connect to the device. Are you sure you are on the same WiFi network? Could there be a firewall blocking things?");
+					} else if (ex.Status == WebExceptionStatus.ConnectionClosed)
+					{
+						dlg.ProgressBox.WriteError("The connection to the device closed unexpectedly. Maybe you tried to do something else with it, or it went to sleep? Please try again.");
```

Comment:
Since there is no way for the user to answer questions. I think it would be better to turn these into statements: E.g.:

Check to ensure that you are on the same wifi network and that there is not a firewall blocking communication.

During communication with HearThis desktop, do not attempt to use the Android device for other things. If the device is going to sleep, you can change the settings to prevent that from happening.